### PR TITLE
Feat: add addon registry update API

### PR DESCRIPTION
**Pull request by chivalryq**
### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/apiserver/rest/webservice/addon_registry.go [hunk 1]
```diff
@@ -70,6 +70,13 @@ func (s *addonRegistryWebService) GetWebService() *restful.WebService {
 		Returns(400, "", bcode.Bcode{}).
 		Writes(apis.AddonRegistryMeta{}))
 
+	ws.Route(ws.POST("/update").To(s.updateAddonRegistry).
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/addon_registry.go:73]: PUT  /{name}

Comment (chivalryq) [reply pkg/apiserver/rest/webservice/addon_registry.go]: done

File: pkg/apiserver/rest/usecase/addon.go [hunk 2]
```diff
@@ -219,6 +227,30 @@ func (u *addonUsecaseImpl) GetAddonRegistryModel(ctx context.Context, name strin
...
+func (u addonUsecaseImpl) UpdateAddonRegistry(ctx context.Context, name string, req apis.CreateAddonRegistryRequest) (*apis.AddonRegistryMeta, error) {
+	var r = model.AddonRegistry{
+		Name: name,
+	}
+	err := u.addonRegistryDS.Get(ctx, &r)
+	if err != nil {
+		return nil, bcode.ErrAddonRegistryNotExist
+	}
+	newRegistry := addonRegistryModelFromCreateAddonRegistryRequest(req)
+	err = u.addonRegistryDS.Delete(ctx, &r)
```
Comment (hongchaodeng) [inline pkg/apiserver/rest/usecase/addon.go:239]: 用 datastore.Put()

Comment (chivalryq) [reply pkg/apiserver/rest/usecase/addon.go]: done

**Comment by chivalryq** [comment]
ping @hongchaodeng @barnettZQG

**Review by barnettZQG — APPROVED** [review]

**Pull request merged**